src/examples/napi_hello/DebugBuild.sh

```diff
@@ -0,0 +1,21 @@
+#! /bin/sh
+# Using GCC on linux ...
+# First build htm.core as debug from sources.
+#      cd <path-to-repo>
+#      mkdir -p build/scripts
+#      cd build/scripts
+#      cmake ../.. -DCMAKE_BUILD_TYPE=Debug
+#      make -j4 install
+#
+# Now build napi_hello
+# The -g -Og tells the compiler to build debug mode with no optimize.
+# We use -std=c++17 to get <filesystem> so we can avoid using the boost library.
+# The -I gives the path to the includes needed to use the htm.core library.
+# The -L gives the path to the shared htm.core library location at build time.
+# -lhtm_core says link with libhtm_core.so
+g++ -g -Og -o napi_hello -std=c++17  -I ../../../build/Debug/include myapp.cpp -L ../../../build/Debug/lib -lhtm_core -lpthread -ldl
```

Comment:
in the final PR, I'm (weakly) against including (specific) build scripts. The readme should be enough, maybe also rewrite this as an example step in the CI (like we have for valgrind etc) and point to it from the readme section too.